### PR TITLE
feat: merging feature branch into master

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Ticket: [INTMDB-1060](https://jira.mongodb.org/browse/INTMDB-1060)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

**Comment by andreaangiolillo** [comment]
FYI: 🔴  tests are failing because of an `OUT_OF_CAPACITY` issue on cloud-dev (we may have run out of ec2 machine available for that region in cloud-dev): [Example](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/6120029790/job/16614215921?pr=1460#step:4:119)

```log
resource_mongodbatlas_federated_database_instance_test.go:109: Step 1/1 error: Error running apply: exit status 1
        
        Error: error creating MongoDB Cluster: POST https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/64fafd3c6cfade3e110312f6/clusters: 409 (request "OUT_OF_CAPACITY") The requested region is currently out of capacity for the requested instance size.
```

File: mongodbatlas/fw_resource_mongodbatlas_encryption_at_rest_test.go [hunk 1]
```diff
@@ -116,28 +117,42 @@ func TestAccAdvRSEncryptionAtRest_basicAWS(t *testing.T) {
...
 			Region:              os.Getenv("AWS_REGION_UPDATED"),
+			RoleID:              os.Getenv("AWS_ROLE_ID"),
 		}
 	)
 
-	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:          func() { testAccPreCheck(t); testCheckAwsEnv(t) },
-		ProviderFactories: testAccProviderFactories,
-		CheckDestroy:      testAccCheckMongoDBAtlasEncryptionAtRestDestroy,
+	resource.Test(t, resource.TestCase{
```
Comment (marcosuma) [inline mongodbatlas/fw_resource_mongodbatlas_encryption_at_rest_test.go:124]: why this is not Parallel anymore?

Comment (andreaangiolillo) [reply mongodbatlas/fw_resource_mongodbatlas_encryption_at_rest_test.go:124]: I am not sure 😅  I will update it. (Wait for a few comments before addressing this one, don't want to run too many times all the tests)

Comment (maastha) [reply mongodbatlas/fw_resource_mongodbatlas_encryption_at_rest_test.go:124]: This was updated because currently these tests do not run on Github pipeline, they can only be run locally. Also, they currently use the same project to manipulate the same resource so these can't be run in parallel even locally. 

The project isn't created inside the test configs as the same project is used for calling update configs as well.

These can be parallel again once we have Azure/GCP credentials setup as well and all tests actually run in actions. Happy to discuss more on this.

File: mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go [hunk 2]
```diff
@@ -854,26 +898,6 @@ resource "mongodbatlas_alert_configuration" "test" {
...
-    threshold   = null
-    units       = null
-    mode        = ""
-  }
-
-  threshold_config {
-    operator  = null
-    threshold = null
-    units     = null
-  }
```
Comment (AgustinBettati) [inline mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go:876]: Would double check on this test. From what I see this was included in master here (https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1419). If we remove the empty matcher and threshold configs not sure if the test has value.

File: mongodbatlas/data_source_mongodbatlas_cloud_provider_access_setup_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
+package mongodbatlas
```
Comment (AgustinBettati) [inline mongodbatlas/data_source_mongodbatlas_cloud_provider_access_setup_test.go]: This file is strange:
- It is over 2 years old
- I do not see it present in master

I imagine we removed it recently and it remained in our migration branch, but cannot find that change in master to get more context.

**Review by AgustinBettati — COMMENTED** [review]
Some details just to double check.
nit: As we are going to merge this branch without squash merge, we could squash the multiple `fixed conflicts` commits into one.

File: mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go (hunk 2 above)
Comment (andreaangiolillo) [reply mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go:876]: I removed the empty check because a field inside the Set became required in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1421 so users cannot add a "null" element in the Set without specifying the required param.

File: mongodbatlas/data_source_mongodbatlas_cloud_provider_access_setup_test.go (hunk 3 above)
Comment (andreaangiolillo) [reply mongodbatlas/data_source_mongodbatlas_cloud_provider_access_setup_test.go]: It was removed here https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1343/files#diff-2637f6d298e2e2a741c6133aa145689eca7a4bf08551a0826bc5f8eaf5543701. the tests were moved to the resource acceptance tests. I will delete this file. Thanks!

Comment (andreaangiolillo) [reply mongodbatlas/data_source_mongodbatlas_cloud_provider_access_setup_test.go]: I removed it

**Review by marcosuma — APPROVED** [review]

File: mongodbatlas/data_source_mongodbatlas_clusters.go [hunk 4]
```diff
@@ -46,14 +46,6 @@ func dataSourceMongoDBAtlasClusters() *schema.Resource {
 							Type:     schema.TypeBool,
 							Computed: true,
 						},
-						"bi_connector": {
```
Comment (maastha) [inline mongodbatlas/data_source_mongodbatlas_clusters.go:49]: Why was this removed?

Comment (andreaangiolillo) [reply mongodbatlas/data_source_mongodbatlas_clusters.go:49]: discussed this offline. Shared this issue via Slack. Waiting for PM's input. Thanks for raising this issue.

Comment (marcosuma) [reply mongodbatlas/data_source_mongodbatlas_clusters.go:49]: I think I did this, I thought it was part of the deprecation removal for 1.12.0?

Comment (andreaangiolillo) [reply mongodbatlas/data_source_mongodbatlas_clusters.go:49]: yes, correct. This is the PR https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1418

**Review by AgustinBettati — APPROVED** [review]
LGTM, just friendly reminder to use regular merge for this one (if possible)

**Review by maastha — APPROVED** [review]
LGTM
